src/LGraphCanvas.ts

```diff
@@ -7872,33 +7875,43 @@ export class LGraphCanvas {
         let targetScale = startScale
         let targetX = startX
         let targetY = startY
+
+        // Find bounding box around nodes
+        let minX = Infinity, minY = Infinity, maxX = -Infinity, maxY = -Infinity
+        nodes.forEach(node => {
+          minX = Math.min(minX, node.pos[0])
+          minY = Math.min(minY, node.pos[1])
+          maxX = Math.max(maxX, node.pos[0] + node.size[0])
+          maxY = Math.max(maxY, node.pos[1] + node.size[1])
+        });
+        const boundingBoxWidth = maxX - minX
+        const boundingBoxHeight = maxY - minY
+        
         if (zoom > 0) {
-            const targetScaleX = (zoom * cw) / Math.max(node.size[0], 300)
-            const targetScaleY = (zoom * ch) / Math.max(node.size[1], 300)
+            const targetScaleX = (zoom * cw) / Math.max(boundingBoxWidth, 300)
+            const targetScaleY = (zoom * ch) / Math.max(boundingBoxHeight, 300)
 
             // Choose the smaller scale to ensure the node fits into the viewport
             // Ensure we don't go over the max scale
             targetScale = Math.min(targetScaleX, targetScaleY, this.ds.max_scale)
-            targetX = -node.pos[0] - node.size[0] * 0.5 + (cw * 0.5) / targetScale
-            targetY = -node.pos[1] - node.size[1] * 0.5 + (ch * 0.5) / targetScale
-        } else {
-            targetX = -node.pos[0] - node.size[0] * 0.5 + (cw * 0.5) / targetScale
-            targetY = -node.pos[1] - node.size[1] * 0.5 + (ch * 0.5) / targetScale
         }
+        targetX = -minX - boundingBoxWidth * 0.5 + (cw * 0.5) / targetScale
+        targetY = -minY - boundingBoxHeight * 0.5 + (ch * 0.5) / targetScale
+        
         const animate = (timestamp: number) => {
             const elapsed = timestamp - startTimestamp
             const progress = Math.min(elapsed / duration, 1)
             const easedProgress = easeFunction(progress)
-
+        
```

Comment:
Remove empty spaces here plz.

Reply:
I wasn't sure whether to use prettier or not, there seems to be a config for it in the repo but the files are not formatted 🤔 For now I've removed those extra lines manually.

Reply:
Oops, just realized you meant the spaces, not the empty lines 😅 restored everything now